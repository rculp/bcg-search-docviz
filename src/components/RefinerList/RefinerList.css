.refiner-list {
  z-index: 0;
}
