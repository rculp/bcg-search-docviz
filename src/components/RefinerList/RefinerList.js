import React from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid/v1';

import { Header } from 'semantic-ui-react';

import './RefinerList.scss';

const RefinerList = ({ box }) => (
  <div className="refiner-list">
    <Header as="h2">{box.display}</Header>
    <ul>
      { box.items.map(item => <li key={uuid()}>{item.Name}</li>) }
    </ul>
  </div>
);

RefinerList.propTypes = {
  box: PropTypes.shape({
    display: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.shape(
      { Name: PropTypes.string }
    ))
  }).isRequired
};

export default RefinerList;
