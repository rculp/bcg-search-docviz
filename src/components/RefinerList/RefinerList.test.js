import React from 'react';
import { shallow } from 'enzyme';
import uuid from 'uuid/v1'; jest.mock('uuid/v1', () => (jest.fn(() => 1)));

import RefinerList from './RefinerList';

describe('Refiner List', () => {
  const props = {
    box: {
      display: 'displayName',
      items: [
        { Name: 'name1' },
        { Name: 'name2' },
        { Name: 'name3' }
      ]
    }
  };

  it('renders and matches our snapshot', () => {
    const component = shallow(
      <RefinerList {...props} />
    );
    expect(component).toMatchSnapshot();
  });
});