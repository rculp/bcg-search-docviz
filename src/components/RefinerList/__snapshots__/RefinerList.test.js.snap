// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Refiner List renders and matches our snapshot 1`] = `
<div
  className="refiner-list"
>
  <Header
    as="h2"
  >
    displayName
  </Header>
  <ul>
    <li
      key="1"
    >
      name1
    </li>
    <li
      key="1"
    >
      name2
    </li>
    <li
      key="1"
    >
      name3
    </li>
  </ul>
</div>
`;
